packages/smithy-aws-core/src/smithy_aws_core/aio/protocols.py

```diff
@@ -0,0 +1,27 @@
+from typing import Literal
+
+from smithy_aws_core.traits import RestJson1Trait
+from smithy_http.aio.protocols import HttpBindingClientProtocol
+from smithy_core.codecs import Codec
+from smithy_core.shapes import ShapeID
+from smithy_json import JSONCodec
+
+
+class RestJsonClientProtocol(HttpBindingClientProtocol):
+    """An implementation of the aws.protocols#restJson1 protocol."""
+
+    _id: ShapeID = RestJson1Trait.id
+    _codec: JSONCodec = JSONCodec()
+    _contentType: Literal["application/json"] = "application/json"
```

Comment:
This isn't necessary, but pro tip:

```suggestion
    _contentType: Final = "application/json"
```

That makes it a `Literal` without having to type it twice.